test/functional/treesitter/testutil.lua

```diff
@@ -0,0 +1,25 @@
+local n = require('test.functional.testnvim')()
+
+local exec_lua = n.exec_lua
+
+local M = {}
+
+---@param language string
+---@param query_string string
+function M.run_query(language, query_string)
```

Comment:
(not a blocker) is this something that would be generally useful in the `vim.treesitter` stdlib?

Reply:
The useful code here would be:

```lua
    local query = vim.treesitter.query.parse(lang, query_str)
    local parser = vim.treesitter.get_parser()
    local tree = parser:parse()[1]
    for id, node, metadata in query:iter_captures(tree:root(), 0) do
       ...
    end
```

And could potentially be used in `runtime/lua/vim/vimhelp.lua`.

But, tbh, I don't think this is useful enough to justify the surface area, as any one of these lines could be changed depending on the application. Do we allow buffer number, lang or parser to be an input, or all 3? Do we want to pass in a query string or a query file path, or both? What kind of iterator do we want to return; iter_matches or iter_captures?